### PR TITLE
Add api for view progress of mstream signal

### PR DESCRIPTION
**Pull request by SPAleksey**

### CODE REVIEW

File: pkg/srv/mstream/event_builder.go [hunk 1]
```diff
@@ -21,17 +21,19 @@ import (
 )
 
 type EventBuilder struct {
-	deviceName     string
-	defragmentedCh <-chan *layers.MStreamFragment
-	writerCh       chan<- []byte
+	deviceName         string
+	defragmentedCh     <-chan *layers.MStreamFragment
+	writerCh           chan<- []byte
+	api_headway_byteCh chan<- []byte
```
Comment (kozhukalov) [inline pkg/srv/mstream/event_builder.go]: please don't use underscores, use instead the camel notation

Comment (SPAleksey) [reply pkg/srv/mstream/event_builder.go]: I am fixed it

File: pkg/srv/mstream/parse_event.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+type AdcData struct {
+	Timestamp uint16
+	Length    uint16
+	Voltage   []uint16
+}
+
+// Encode []byte to struct Mstrean signal doc from TQDC2_Data_Format_rev0.pdf and tqdc16vs.py
+func MstreamHeaderConstructor(d []byte) MstreamHeader {
```
Comment (kozhukalov) [inline pkg/srv/mstream/parse_event.go]: It is a convention to name constructor methods as NewSomthing

File: pkg/srv/mstream/mstream.go [hunk 3]
```diff
@@ -170,6 +176,9 @@ func (s *MStreamServer) Run() error {
 			}
 		}(s.writerStateChs[deviceName], s.writerChs[deviceName])
 
+		//Run parse event from event builders for api
+		go ParseEventsForApi(s.apiProgressByteCh, s.apiProgressJsonCh, s.apiProgressReqCh)
```
Comment (kozhukalov) [inline pkg/srv/mstream/mstream.go]: You run multiple parsers (one per device), but all event builders put event data in the same channel.

File: pkg/srv/mstream/api.go [hunk 4]
```diff
@@ -132,6 +132,16 @@ func (s *ApiServer) configureRouter() {
...
+	// swagger:operation GET /headway mstream getHeadway
+	// ---
+	// summary: headway mstream
+	// description: --
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/okResp"
+	//   "400":
+	//     "$ref": "#/responses/badReq"
+	subRouter.HandleFunc("/progress", s.handleProgress()).Methods("GET")
```
Comment (kozhukalov) [inline pkg/srv/mstream/api.go]: I believe this api handler must per device. Smth like /progress/<device_name>

File: pkg/srv/mstream/event_builder.go [hunk 5]
```diff
@@ -24,14 +25,16 @@ type EventBuilder struct {
 	deviceName     string
 	defragmentedCh <-chan *layers.MStreamFragment
 	writerCh       chan<- []byte
+	apiHeadwayCh   chan<- []byte
```
Comment (kozhukalov) [inline pkg/srv/mstream/event_builder.go]: Let's use similar name as in the mstream server module.

File: pkg/srv/mstream/api.go (hunk 4 above)
Comment (kozhukalov) [reply pkg/srv/mstream/api.go]: Even more it is rather /last_event/<device_name>

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,4 @@
 .idea
 .vscode
 bin
+go-adc.exe
```
Comment (kozhukalov) [inline .gitignore]: newline

File: pkg/srv/mstream/parse_event.go (hunk 2 above)
Comment (SPAleksey) [reply pkg/srv/mstream/parse_event.go]: I am fixed it

File: pkg/srv/mstream/mstream.go (hunk 3 above)
Comment (SPAleksey) [reply pkg/srv/mstream/mstream.go]: I am fixed it

File: pkg/srv/mstream/api.go (hunk 4 above)
Comment (SPAleksey) [reply pkg/srv/mstream/api.go]: I am fixed it

File: pkg/srv/mstream/event_builder.go (hunk 5 above)
Comment (SPAleksey) [reply pkg/srv/mstream/event_builder.go]: I am fixed it

File: .gitignore (hunk 6 above)
Comment (SPAleksey) [reply .gitignore]: I am fixed it

File: pkg/srv/mstream/api.go [hunk 7]
```diff
@@ -166,6 +176,13 @@ func (s *ApiServer) handleFlush() http.HandlerFunc {
 	}
 }
 
+func (s *ApiServer) handleLastEvent() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Debug("Handling last event request for device ", mux.Vars(r)["deviceName"])
+		w.Write(MstreamHeaderJson(<-s.mstream.lastEventChs[mux.Vars(r)["deviceName"]]))
```
Comment (kozhukalov) [inline pkg/srv/mstream/api.go]: Please use here smth like this
```
if val, ok := s.mstream.lastEventChs[mux.Vars(r)["deviceName"]]]; ok {
    //do smth useful
} else {
   http.Error(w, err.Error(), http.StatusNotFound)
}
```

Comment (SPAleksey) [reply pkg/srv/mstream/api.go]: I am fixed it

**Comment by SPAleksey** [comment]
The case of API is not handled if mstream has started but no data has been received into the channel (lastEventCh) at least once

**Review by kozhukalov — APPROVED** [review]

**Pull request merged**